go/workflow.go

```diff
@@ -0,0 +1,333 @@
+			fmt.Sprintf(EmailWelcome, customer.StatusLevel.Name)).
+			Get(ctx, nil)
+		if err != nil {
+			logger.Error("Error running SendEmail activity.", "Error", err)
+			return err
+		}
+	}
+
+	// signal handler for adding points
+	selector.AddReceive(workflow.GetSignalChannel(ctx, SignalAddPoints),
+		func(c workflow.ReceiveChannel, _ bool) {
+			errSignal = signalAddPoints(ctx, c, &customer)
+		})
+
+	// signal handler for adding guest
+	selector.AddReceive(workflow.GetSignalChannel(ctx, SignalInviteGuest),
+		func(c workflow.ReceiveChannel, _ bool) {
+			errSignal = signalInviteGuest(ctx, c, &customer)
+		})
+
+	// signal handler for ensuring the customer is at least the given status. Used for invites and promoting an existing account.
+	selector.AddReceive(workflow.GetSignalChannel(ctx, SignalEnsureMinimumStatus),
+		func(c workflow.ReceiveChannel, _ bool) {
+			errSignal = signalEnsureMinimumStatus(ctx, c, &customer)
+		})
+
+	// signal handler for canceling account
+	selector.AddReceive(workflow.GetSignalChannel(ctx, SignalCancelAccount),
+		func(c workflow.ReceiveChannel, _ bool) {
+			errSignal = signalCancelAccount(ctx, c, &customer)
+		})
+
+	// handle Temporal Server cancellation requests
+	selector.AddReceive(ctx.Done(),
+		func(c workflow.ReceiveChannel, _ bool) {
+			c.Receive(ctx, nil)
+			logger.Info("Cancellation requested.")
+			workflowCanceled = true
+		})
+
+	// query handler for status level, etc
+	err = workflow.SetQueryHandler(ctx, QueryGetStatus,
+		func() (GetStatusResponse, error) {
+			return queryGetStatus(ctx, customer)
+		})
+	if err != nil {
+		return fmt.Errorf("unable to register '%v' query handler: %w", QueryGetStatus, err)
+	}
+
+	// query handler for guest list
+	err = workflow.SetQueryHandler(ctx, QueryGetGuests,
+		func() ([]string, error) {
+			return queryGetGuests(ctx, customer)
+		})
+	if err != nil {
+		return fmt.Errorf("unable to register '%v' query handler: %w", QueryGetGuests, err)
+	}
+
+	// Block on everything. Continue-As-New on history length; size of activities in this workflow are small enough
+	// that we'll hit the length thresholds well before any size threshold.
+	logger.Info("Waiting for new signals")
+	for customer.AccountActive && info.GetCurrentHistoryLength() < EventsThreshold && !workflowCanceled {
+		selector.Select(ctx)
+
+		if errSignal != nil {
+			logger.Error("Unrecoverable error in handling a signal.", "Error", errSignal)
+			return errSignal
+		}
+	}
+
+	// here because of events threshold, but account still active? Continue-As-New
+	if customer.AccountActive && !workflowCanceled {
+		return workflow.NewContinueAsNewError(ctx, customer)
+	}
+
+	logger.Info("Loyalty workflow completed.", "Customer", customer, "WorkflowCanceled", workflowCanceled)
+	return nil
+}
+
+// CustomerWorkflowID generates a Workflow ID based on the given customer ID.
+func CustomerWorkflowID(customerID string) string {
+	return fmt.Sprintf("customer-%v", customerID)
```

Comment:
```suggestion
	return "customer- + customerID
```

Can just use basic concatenation in these simple cases, but your choice